### PR TITLE
Added Hash of MAC address for nearby bluetooth devices

### PR DESCRIPTION
**Pull request by this-Aditya**

### CODE REVIEW

File: plugins/radar-android-phone/src/main/java/org/radarbase/passive/phone/PhoneBluetoothManager.kt [hunk 1]
```diff
@@ -79,20 +105,59 @@ class PhoneBluetoothManager(service: PhoneBluetoothService) : AbstractSourceMana
...
+
+                            val scannedTopicBuilder = PhoneBluetoothDeviceScanned.newBuilder().apply {
+                                time = currentTime
+                                timeReceived = currentTime
+                            }
+
+                            val pairedDevices: Set<BluetoothDevice> = if (hasConnectPermission) bluetoothAdapter.bondedDevices else emptySet()
+                            val newPairedDevices = pairedDevices.subtract(allPairedDevices)
+                            allPairedDevices.addAll(newPairedDevices)
+                            allPairedDevices.removeAll(allPairedDevices.subtract(pairedDevices)) // removing devices which are previously paired but unpaired now
```
Comment (yatharthranjan) [inline plugins/radar-android-phone/src/main/java/org/radarbase/passive/phone/PhoneBluetoothManager.kt:136]: isn't this essentially just keeping the current pairedDevices in allPairedDevices? So why do we do all the subtract/add ?

Comment (yatharthranjan) [reply plugins/radar-android-phone/src/main/java/org/radarbase/passive/phone/PhoneBluetoothManager.kt:136]: why not just keep the latest list in that case? `allPairedDevices = pairedDevices`

File: gradle.properties [hunk 2]
```diff
@@ -16,6 +16,7 @@ org.gradle.jvmargs=-Xmx3072m -XX:MaxMetaspaceSize=1536m -Dfile.encoding=UTF-8 -X
 org.gradle.parallel=true
 org.gradle.vfs.watch=true
 kotlin.code.style=official
+android.jetifier.ignorelist=jackson-core-2.16.1.jar
```
Comment (this-Aditya) [inline gradle.properties:19]: CI tests are failing after updating the schemas to the latest version. Jackson-core is responsible for the failing tests. Hence, I found [this](https://issuetracker.google.com/issues/322403838#comment5) solution to ignore it from Jetifier.

**Review by yatharthranjan — APPROVED** [review]
LGTM, thanks

**Comment by yatharthranjan** [comment]
Hi @this-Aditya just disussing this today. Is the salting really needed on the hash? As it might change (due to reset/reinstall of thw app) and then the same device will not have the same hash making analysis difficult.  I think since it is just a MAC address (and not something sensitive like a password), perhaps we can just use hashing without the salt.

If really needed, we should put it in remote config (qualified by projectId and userId) so that it does not reset.

**Pull request merged**